qa/L0_resnet50_ensemble/test_client.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/env python
+
+# The MIT License (MIT)
+#
+# Copyright (c) 2020-2025 NVIDIA CORPORATION & AFFILIATES
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+# the Software, and to permit persons to whom the Software is furnished to do so,
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import argparse, os, sys
+import numpy as np
+from numpy.random import randint
+import tritongrpcclient
+from PIL import Image
+
+np.random.seed(100019)
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-v', '--verbose', action="store_true", required=False, default=False,
+                        help='Enable verbose output')
+    parser.add_argument('-u', '--url', type=str, required=False, default='localhost:8001',
+                        help='Inference server URL. Default is localhost:8001.')
+    parser.add_argument('--batch_size', type=int, required=False, default=1,
+                        help='Batch size')
+    parser.add_argument('--n_iter', type=int, required=False, default=-1,
+                        help='Number of iterations , with `batch_size` size')
+    parser.add_argument('--model_name', type=str, required=False, default="dali_backend",
+                        help='Model name')
+    img_group = parser.add_mutually_exclusive_group()
+    img_group.add_argument('--img', type=str, required=False, default=None,
+                           help='Run a img dali pipeline. Arg: path to the image.')
+    img_group.add_argument('--img_dir', type=str, required=False, default=None,
+                           help='Directory, with images that will be broken down into batches and inferred. '
+                                'The directory must contain only images and single labels.txt file')
+    return parser.parse_args()
+
+
+def load_image(img_path: str):
+    """
+    Loads image as an encoded array of bytes.
+    This is a typical approach you want to use in DALI backend
+    """
+    with open(img_path, "rb") as f:
+        img = f.read()
+        return np.array(list(img)).astype(np.uint8)
```

Comment:
```suggestion
    return np.fromfile(img_path, dtype=np.uint8)
```

Reply:
Absolutely! It's literally orders of magnitude faster.